### DISCUSSION TITLE
External oauth (discord etc)

### DISCUSSION THREAD

**Discussion by cheesycod** [piccolo-orm/piccolo_admin#454] [General]
Would be nice to be able to use discord oauth or another external auth system

**Comment by dantownsend** [comment]
@cheesycod It's a nice idea. My knowledge of OAuth is quite limited. I need to find out if there's an existing async OAuth library for Python which could be integrated.

**Comment by sinisaos** [comment]
@dantownsend There is library [Authlib](https://docs.authlib.org/en/latest/client/fastapi.html) but I don't think that's within the scope of Piccolo Admin because the admin interface is for administrators, not for the end user. Usually administrator interfaces have a login with a username and password, and Piccolo Admin already has that.

**Comment by cheesycod** [comment]
There are many people who use oauth internally including large companies like google etc.

if not oauth, maybe it could be possible to create a custom auth function or smth?
